include/wil/registry.h

```diff
@@ -17,16 +17,2704 @@
+            regview.create_key(path, &return_value, access);
+            return return_value;
+        }
+
+#if defined(__WIL_WINREG_STL)
+        /**
+         * \brief Opens a new HKEY to the specified path - see RegOpenKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param access The requested access desired for the opened key
+         * \return A wil::shared_hkey containing the resulting opened HKEY
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures
+         */
+        inline ::wil::shared_hkey open_shared_key(HKEY key, PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            ::wil::shared_hkey return_value;
+            regview.open_key(path, &return_value, access);
+            return return_value;
+        }
+
+        /**
+         * \brief Creates a new HKEY to the specified path - see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param access The requested access desired for the opened key
+         * \return A wil::shared_hkey or wil::shared_hkey containing the resulting opened HKEY
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures
+         */
+        inline ::wil::shared_hkey create_shared_key(HKEY key, PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+        {
+            const reg_view_details::reg_view regview{ key };
+            ::wil::shared_hkey return_value;
+            regview.create_key(path, &return_value, access);
+            return return_value;
+        }
+#endif // #if defined(__WIL_WINREG_STL)
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
+
+        /**
+         * \brief Opens a new HKEY to the specified path - see RegOpenKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param[out] hkey A reference to a wil::unique_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT open_unique_key_nothrow(HKEY key, _In_opt_ PCWSTR path, _Out_::wil::unique_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT

```

Comment:
this is `_Inout_`, but you don't need the SAL to say that.
drop the SAL. but if you really want to keep it, add a space after the annotation.

Reply:
yeah. I added it just as a hint to the developer that we'll write to it. fixing.